specutils/io/default_loaders/sdss_v.py

```diff
@@ -0,0 +1,555 @@
+"""Register reader functions for various spectral formats."""
+from collections import OrderedDict
+from typing import Optional
+
+import numpy as np
+from astropy.units import Unit, Quantity, Angstrom
+from astropy.nddata import StdDevUncertainty, InverseVariance
+from astropy.io.fits import HDUList
+
+from ...spectra import Spectrum1D, SpectrumList
+from ..registers import data_loader
+from ..parsing_utils import read_fileobj_or_hdulist
+
+__all__ = [
+    "load_sdss_apVisit_1D",
+    "load_sdss_apVisit_list",
+    "load_sdss_apStar_1D",
+    "load_sdss_apStar_list",
+    "load_sdss_spec_1D",
+    "load_sdss_spec_list",
+    "load_sdss_mwm_1d",
+    "load_sdss_mwm_list",
+]
+
+
+## IDENTIFIER
+def identify_filetype(filetype):
+    """Generic function generator to verify if a file_obj string specificies this filetype."""
+    return lambda f, o, *args, **kwargs: o.split("/")[-1].startswith(filetype)
+
+
+## HELPERS
+def _wcs_log_linear(naxis, cdelt, crval):
+    """
+    Convert WCS from log to linear.
+    """
+    return 10**(np.arange(naxis) * cdelt + crval)
+
+
+def _fetch_metadata(hdulist):
+    """
+    Fetch the relevant common metadata.
+
+    Global function used in all SDSS-V metadata handling.
+
+    Parameters
+    ----------
+    hdulist: HDUList
+        The hdulist imported from fits.open().
+
+    Returns
+    -------
+    meta: OrderedDict
+        Dictionary of relevant metadata from primary HDU.
+
+    """
+    # Orderly with an OrderedDict
+    common_meta = OrderedDict([])
+
+    # Loop over all keys to get relevant metadata
+    for key in hdulist[0].header.keys():
+        # exclude these keys
+        if key.startswith(("TTYPE", "TFORM", "TDIM")) or key in (
+                "",
+                "COMMENT",
+                "CHECKSUM",
+                "DATASUM",
+                "NAXIS",
+                "NAXIS1",
+                "NAXIS2",
+                "XTENSION",
+                "BITPIX",
+                "PCOUNT",
+                "GCOUNT",
+                "TFIELDS",
+        ):
+            continue
+
+        common_meta[key.lower()] = hdulist[0].header.get(
+            key)  # add key to dict
```

Comment:
Also see discussions in https://github.com/astropy/specutils/issues/617 and https://github.com/astropy/specutils/issues/1102

Reply:
[b61a668](https://github.com/astropy/specutils/pull/1107/commits/b61a668a7eec8b362e4a7498dfe490c2c5bcb10a)

This should be fine then. I haven't tested the tabular writer `wcs1d_fits_writer` , but I'd assume leaving this header as is in the metadata would be fine. 

`mwmVisit/Star` might fail on writing like that though, since it partially stores some metadata within the `data` component of the `BinTableHDU` (SNR, telescope, observation date, MJD), so I've left that in the `meta` dictionary for now.

Reply:
I'm not too worried about people writing the `Spectrum1D` objects back to files.   Currently, both writers would write out files that are not in the original format, and not that useful.  Many of the SDSS products in here may not write correctly at the moment.   

There is a longer ongoing discussion about improving the default writers to make more fits-like things.  We could write custom writers for our loaders, but I think that's out of scope for this PR.